tools/cli/commands/project/create.js

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Coherent Labs AD. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const { exec } = require('node:child_process');
+
+const supportedTemplates = ['webpack', 'vite'];
+
+/**
+ * Validate if the passed template is valid
+ * @param {string} template the name of the template
+ * @returns {boolean}
+ */
+function validateTemplate(template) {
+    const valid = supportedTemplates.includes(template.toLowerCase());
+
+    if (!valid) console.log(chalk.red(`Template ${template} is not a valid option. Please use one of the supported templates - ${supportedTemplates.join(', ')}!`));
+
+    return valid;
+}
+
+/**
+ * Validate if the passed directory exists
+ * @param {string} directory the path to the directory
+ * @returns {string|null[]} tuple - error, path
+ */
+function setupDirectory(directory) {
+    const fullPath = path.resolve(process.cwd(), directory);
+    const exists = fs.existsSync(fullPath);
+
+    if (!exists) {
+        fs.mkdirSync(fullPath);
+        return [null, fullPath];
+    }
+
+    const content = fs.readdirSync(fullPath);
+    if (!content.length) return [null, fullPath];
+
+    const error = `ERROR: Directory ${directory} is not empty.`;

```

Comment:
Enchancement idea: We can prompt the user if they want to have the folder cleared and use it.